qualtran/cirq_interop/_cirq_interop.py

```diff
@@ -426,38 +417,85 @@ def _cbloq_to_cirq_circuit(
         circuit: The cirq.FrozenCircuit version of this composite bloq.
         cirq_quregs: The output mapping from right register names to Cirq qubit arrays.
     """
-    soq_assign: Dict[Soquet, CirqQuregT] = {}
-    _update_assign_from_cirq_quregs(signature.lefts(), LeftDangle, cirq_quregs, soq_assign)
+    cirq_quregs = {k: np.apply_along_axis(_QReg, -1, v) for k, v in cirq_quregs.items()}
+    qvar_to_qreg: Dict[Soquet, _QReg] = {
+        Soquet(LeftDangle, idx=idx, reg=reg): cirq_quregs[reg.name][idx]
+        for reg in signature.lefts()
+        for idx in reg.all_idxs()
+    }
     moments: List[cirq.Moment] = []
     for binsts in nx.topological_generations(binst_graph):
         moment: List[cirq.Operation] = []
 
         for binst in binsts:
-            if isinstance(binst, DanglingT):
+            if binst == LeftDangle:
```

Comment:
elsewhere I've been using `binst is LeftDangle`

Reply:
changed.